hypervisor/src/lib.rs

```diff
@@ -187,3 +188,98 @@ pub enum IrqRoutingEntry {
     #[cfg(feature = "mshv")]
     Mshv(mshv_bindings::mshv_msi_routing_entry),
 }
+
+#[derive(Debug, Clone, Copy, PartialEq, serde::Serialize, serde::Deserialize)]
+pub enum StandardRegisters {
+    #[cfg(feature = "kvm")]
+    Kvm(kvm_bindings::kvm_regs),
+    #[cfg(all(feature = "mshv", target_arch = "x86_64"))]
+    Mshv(mshv_bindings::StandardRegisters),
+}
+
+impl Default for StandardRegisters {
+    fn default() -> Self {
+        #[cfg(feature = "kvm")]
+        return kvm_bindings::kvm_regs::default().into();
+        #[cfg(feature = "mshv")]
+        #[allow(unreachable_code)]
+        {
+            mshv_bindings::StandardRegisters::default().into()
+        }
+    }
+}
```

Comment:
I don't see why you need such Default trait.

Reply:
There are tests in Emulator which relies on the default implementation.

Reply:
The default implementation does not look right to me, since it will always default to kvm registers when both kvm and mshv are enabled. This can use mis-use in the future.

If there are tests need to construct default `StandardRegisters`, use mshv or kvm register default trait with the `.into()` `StandardRegisters` and guard it with the appropriate feature gates.